_posts/2019-07-16-gentle-intro-to-docker.markdown

```diff
@@ -0,0 +1,65 @@
+
+# And we widen to reveal....
+
+Enter [Docker](http://docker.com).  It started life as a cloud computing platform, but after a few years, it's creators open sourced it. Open sourcing it did what open sourcing does - changed things - and now we have the modern Docker you've been hearing so much about. But Docker does more than just generalizing the abilities of a language specific manager to any executable application - it brings a number of new abilities to the table. You can migrate a container to an entirely new host, replicate a single container any arbitrary number of times, and encapsulate complex services in an easy to run, atomic fashion. A full listing could (and has) fill a book.  You get the idea.
+
+# Show me
+
+Glad you asked.  The simplest way to get started with Docker is to [install it](https://hub.docker.com/) - you'll have to create an account to get the goodies, and the installation can be a bit of a process, so be ready for that.  Once the smoke has cleared and you're done gritting your teeth, open up a command line and try:
+
+``docker run hello-world``
+
+Neat, if a little underwhelming.  Read the output, by the way - it has a decent explanation of how Docker works in general - your command downloaded an image, then created a container for the image, ran some commands and captured their output for you.  Note that an image and a container are two separate concepts - you can have one image, but have it running in as many different containers as you want.  The image is the template used to create containers, much like a class is a template used to created individual objects of that class.
+
+How about something that's a little more fun to play with?  Run:
+
+``docker run -it ubuntu bash``
+
+Just like before, Docker will download an image - this time, the well known Linux variant Ubuntu - and then create a container to run it in.  Note the extra argument that we added at the end, though - a command that will be executed within the downloaded image.  It's the same basic syntax (and idea) that you would use to run a command remotely with ssh.  In this case, it'll give you a nice, interactive shell that lets you poke around in your container.  The ``-it`` bit is a pair of command line options that allocate a terminal and tell Docker that you want to actually interact with the container.  Now try:
```

Comment:
> a command that will be executed within the downloaded image.

`a command that will be executed within the new created container.` might be better, what you think?